packages/backend-common/src/reading/GoogleGcsUrlReader.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Storage } from '@google-cloud/storage';
+import {
+  ReaderFactory,
+  ReadTreeResponse,
+  SearchResponse,
+  UrlReader,
+} from './types';
+import getRawBody from 'raw-body';
+import {
+  GOOGLE_GCS_HOST,
+  readGoogleGcsIntegrationConfig,
+} from '@backstage/integration';
+
+const parseURL = (
+  url: string,
+): { host: string; bucket: string; key: string } => {
+  const { host, pathname } = new URL(url);
+
+  if (host !== GOOGLE_GCS_HOST) {
+    throw new Error(`not a valid GCS URL: ${url}`);
+  }
+
+  const [, bucket, ...key] = pathname.split('/');
+  return {
+    host: host,
+    bucket,
+    key: key.join('/'),
+  };
+};
+
+export class GoogleGcsUrlReader implements UrlReader {
+  static factory: ReaderFactory = ({ config, logger }) => {
+    if (!config.has('integrations.googleGcs')) {
+      return [];
+    }
+    const gcsConfig = readGoogleGcsIntegrationConfig(
+      config.getConfig('integrations.googleGcs'),
+    );
+    let storage: Storage;
+    if (!gcsConfig.clientEmail || !gcsConfig.privateKey) {
+      logger.warn(
+        'googleGcs credentials not found in config. Using default credentials provider.',
```

Comment:
I'd suggest removing this and instead have `toString` show whether it's relying on explicit auth or not. All installed UrlReaders are logged at startup, so it'll be visible there.

Reply:
I'm not sure how would I know in toString about the credentials that were provided. Do you want me to add another variable that indicates that? 

Also, I have used this pattern in another plugin that requres aws credentials and I must say it's been really helpful to distinguish when it was using the explicitly provided credentials vs the implicit ones found by the SDK.

Happy to change it if you'd still prefer this to not be here.

Reply:
Guess you could check if there's something available on the `Storage` instance to avoid having to pass a flag in. Perhaps keep the logging here then but make it an `info`? Just a bit to odd to warn about expected behaviour